src/inference/include/openvino/runtime/config.hpp

```diff
@@ -0,0 +1,380 @@
+     * @param str_ configuration key string
+     */
+    constexpr Key(const char* str_) : _str{str_} {}
+
+    /**
+     * @brief Constructs configuration
+     * @tparam Args configuration paramter constructor arguments types
+     * @param args configuration paramter constructor arguments
+     * @return Pair of string key representation and type erased paramter.
+     */
+    template <typename... Args>
+    std::pair<std::string, Any> operator()(Args&&... args) const {
+        return {_str, Any::make<T>(std::forward<Args>(args)...)};
+    }
+
+    /**
+     * @brief return configuration key string
+     * @return Pointer to const string key representation
+     */
+    const char* str() const {
+        return _str;
+    }
+
+private:
+    const char* _str = nullptr;
+};
+
+/**
+ * @brief Metric to get a std::vector<std::string> of available device IDs
+ */
+static constexpr Key<std::vector<std::string>> available_devices{"AVAILABLE_DEVICES"};
+
+/**
+ * @brief Metric to get a std::vector<std::string> of supported metrics.
+ *
+ * This can be used as an executable network metric as well.
+ *
+ * Each of the returned device metrics can be passed to Core::GetMetric, executable network metrics
+ * can be passed to ExecutableNetwork::GetMetric.
+ *
+ */
+static constexpr Key<std::vector<std::string>> supported_metrics{"SUPPORTED_METRICS"};
+
+/**
+ * @brief Metric to get a std::vector<std::string> of supported config keys.
+ *
+ * This can be used as an executable network metric as well.
+ *
+ * Each of the returned device configuration keys can be passed to Core::SetConfig, Core::GetConfig, and
+ * Core::LoadNetwork, configuration keys for executable networks can be passed to ExecutableNetwork::SetConfig and
+ * ExecutableNetwork::GetConfig.
+ *
+ */
+static constexpr Key<std::vector<std::string>> supported_config_keys{"SUPPORTED_CONFIG_KEYS"};
+
+/**
+ * @brief Metric to get a std::string value representing a full device name.
+ */
+static constexpr Key<std::string> full_device_name{"FULL_DEVICE_NAME"};
+
+/**
+ * @brief Metric to get a std::vector<std::string> of optimization options per device.
+ *
+ */
+static constexpr Key<std::vector<std::string>> optimization_capabilities{"OPTIMIZATION_CAPABILITIES"};
+namespace capabilities {
+static constexpr auto fp32 = "FP32";  //!< device can support FP32 models
+static constexpr auto bf16 = "BF16";  //!< device can support BF16 computations for models
+static constexpr auto fp16 = "FP16";  //!< device can support FP16 models
+static constexpr auto int8 = "int8";  //!< device can support models with INT8 layers
+static constexpr auto bin = "bin";    //!< device can support models with BIN layers
+static constexpr auto winograd =
+    "WINOGRAD";  //!< device can support models where convolution implemented via Winograd transformations
+static constexpr auto batched_blob = "BATCHED_BLOB";  //!< device can support BatchedBlob
+}  // namespace capabilities
```

Comment:
Why it is not an enum?

Reply:
removed string enumeration. Seems it is more documentation section with out incode usage